injected/src/features/broker-protection/captcha-services/providers/image.js

```diff
@@ -0,0 +1,80 @@
+import { PirError } from '../../types';
+import { svgToBase64Jpg, imageToBase64 } from '../utils/image';
+import { injectTokenIntoElement } from '../utils/token';
+import { isElementType } from '../utils/element';
+import { stringifyFunction } from '../utils/stringify-function';
+
+/**
+ * @import { CaptchaProvider } from './provider.interface';
+ * @implements {CaptchaProvider}
+ */
+export class ImageProvider {
+    getType() {
+        return 'image';
+    }
+
+    /**
+     * @param {HTMLElement} captchaImageElement - The captcha image element
+     */
+    isSupportedForElement(captchaImageElement) {
+        if (!captchaImageElement) {
+            return false;
+        }
+
+        return isElementType(captchaImageElement, ['img', 'svg']);
+    }
+
+    /**
+     * @param {HTMLElement} captchaImageElement - The captcha image element
+     */
+    async getCaptchaIdentifier(captchaImageElement) {
+        if (captchaImageElement.tagName.toLocaleLowerCase() === 'svg') {
+            // Converting to unknown to avoid TypeScript error when going from HTMLElement to SVGElement
+            const unknownElement = /** @type {unknown} */ (captchaImageElement);
+            const svgElement = /** @type {SVGElement} */ (unknownElement);
+            const base64Image = await svgToBase64Jpg(svgElement);
```

Comment:
nit: To avoid this you can create a type guard:

```js
/**
 * @param {HTMLElement} element
 * @return {element is SVGElement}
 */
function isSVGElement(element) {
  return element.tagName.toLocaleLowerCase() === 'svg';
}
```

and use as follows:
```suggestion
        if (isSVGElement(captchaImageElement)) {
            const base64Image = await svgToBase64Jpg(captchaImageElement);
```

Reply:
Ah, that’s much better. Thank you!